app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,49 @@
+class HallsController < ApplicationController
+before_action :set_hall, only: %i[ show edit update destroy ]
```

Comment:
I don't see an action named `show` in your controller, so you should either remove it from the list or create such an action.